pkg/client/elasticsearch/client.go

```diff
@@ -105,6 +105,24 @@ type MetricsClient struct {
 	logger logr.Logger
 }
 
+// ErrorResponse is an Elasticsearch error response.
+type ErrorResponse struct {
+	Status int `json:"status"`
+	Error  struct {
+		CausedBy struct {
+			Reason string `json:"reason"`
+			Type   string `json:"type"`
+		} `json:"caused_by"`
+		Reason     string `json:"reason"`
+		Type       string `json:"type"`
+		StackTrace string `json:"stack_trace,omitempty"`
+		RootCause  []struct {
+			Reason string `json:"reason"`
+			Type   string `json:"type"`
+		} `json:"root_cause"`
+	} `json:"error"`
+}
+
```

Comment:
We can simplify the struct to only contain the fields we are interested in (https://github.com/elastic/elasticsearch-k8s-metrics-adapter/pull/64#discussion_r1379874296).

```suggestion
// ErrorResponse is an Elasticsearch error response.
type ErrorResponse struct {
	Error  struct {
		Reason     string `json:"reason"`
		Type       string `json:"type"`
	} `json:"error"`
}

```

Reply:
I can make that change, I was just wondering if we want the extra fields in case we need it in the future.